### PR TITLE
Adding StringExpression api class for bq-aggn-pushdown

### PR DESCRIPTION
**Pull request by sanjanasandeep**

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/relational/StringExpression.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    public String getExpression() {
+        return expression;
+    }
+
+    /**
+     * A {@link StringExpression} is always assumed to be valid. It is the responsibility of the creator of the
+     * object to ensure correctness of the string expression.
+     * @return This method always returns true.
+     */
```
Comment (sanjanasandeep) [inline cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/relational/StringExpression.java:37]: is this assumption valid?

Comment (fernst) [reply cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/relational/StringExpression.java:37]: Yes

**Review by fernst — APPROVED** [review]
LGTM

**Review by DJSagarAhire — COMMENTED** [review]
Why do we need to add this? I'm adding a `SQLExpression` and `SQLExpressionFactory` class in google-cloud after it was decided to remove the classes from cdap.

**Comment by fernst** [comment]
> Why do we need to add this? I'm adding a `SQLExpression` and `SQLExpressionFactory` class in google-cloud after it was decided to remove the classes from cdap.

Because the Plugins don't have a dependency to the Google Cloud project. These classes need to be present in the main CDAP packages so all plugins (Core Plugins, Google Cloud plugins) can use the same common classes to "interact" with each other.

File: cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/relational/StringExpression.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
```
Comment (sanjanasandeep) [inline cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/relational/StringExpression.java:2]: @DJSagarAhire  Adding this class as mentioned in the design doc: https://docs.google.com/document/d/1hYnP_jAday-RKI0TPkJX_r2iJeroZXZAZB8XWx1nTDI/edit?resourcekey=0-lS0CoG1Ji02whjnCwRZJ_w#. @fernst and I synced, and it was decided that we need this class in cdap, to construct SQL expression in BigQuerySQLEngine

Comment (DJSagarAhire) [reply cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/relational/StringExpression.java:2]: Got it. So does this replace the Expression / ExpressionFactory classes that are being added in google-cloud?

**Review by tivv — CHANGES_REQUESTED** [review]
This must not be in api

**Comment by sanjanasandeep** [comment]
closing this as api is not needed

**Pull request closed without merging**